contracts/account/paymaster/PaymasterNFT.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC721} from "@openzeppelin/contracts/interfaces/IERC721.sol";
+import {ERC4337Utils, PackedUserOperation} from "@openzeppelin/contracts/account/utils/draft-ERC4337Utils.sol";
+import {PaymasterCore} from "./PaymasterCore.sol";
+
+/**
+ * @dev Extension of {PaymasterCore} that supports account based on ownership of an ERC-721 token
+ */
+abstract contract PaymasterNFT is PaymasterCore {
```

Comment:
I think `PaymasterNFT` was a mistake, and we should've called it `PaymasterERC721`, did we discussed it? It's just that now it seems pretty arbitrary when reviewing 😅

Reply:
That was the name in the grant agreement.

I'd be ok with changing it, but I would go even further and call it something like `PaymasterERC721Owner` to clarify that the ownership gives you the right and you are not actually giving away the NFT as a payment.